source/common/grpc/async_client_impl.h

```diff
@@ -61,6 +61,7 @@ class AsyncClientImpl final : public AsyncClient<RequestType, ResponseType> {
   }
 
 private:
+  Http::MessagePtr headers_msg_;
```

Comment:
I think this needs to be in the stream? (The client can start multiple streams) This is similar to what the raw HTT async client impl does.